### PR TITLE
Graphql

### PR DESCRIPTION
**Pull request by KMosc**
I'm working on it.
- [x] Add graphql
- [x] Add resolvers

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,7 @@ source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '3.0.0'
+gem 'graphql'
```
Comment (wmaciejak) [inline Gemfile:5]: Add gem version ;)

Comment (KMosc) [reply Gemfile:5]: Right, You mentioned it on feedback.

File: app/graphql/mutations/cinema_hall_mutation.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Mutations
+    class CinemaHallMutation < BaseMutation
+      argument :volume, Integer, required: false
+      
+      def resolve(**args)
+        repository = Repository::CinemaHallRepository.new
+        UseCase::CinemaHalls::Create.new(repository).call(args)
+        {status: 200}
```
Comment (wmaciejak) [inline app/graphql/mutations/cinema_hall_mutation.rb]: Consider using Rubocop, here you have missing spaces

Comment (wmaciejak) [reply app/graphql/mutations/cinema_hall_mutation.rb]: Also (I'm not sure about this one cause last time I've used GraphQL years ago), you don't want to return created object in response?

File: app/graphql/mutations/movie_mutation.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module Mutations
+    class MovieMutation < BaseMutation
+      argument :title, String, required: false
+      argument :description, String, required: false
+      argument :age_restriction, Integer, required: false
+      argument :genre_id, ID, required: false
```
Comment (wmaciejak) [inline app/graphql/mutations/movie_mutation.rb]: Genre is not required for movie?

File: app/graphql/mutations/movie_mutation.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module Mutations
+    class MovieMutation < BaseMutation
```
Comment (wmaciejak) [inline app/graphql/mutations/movie_mutation.rb]: Consider naming convention - CreateMovieMutation, cause you will have separated Update and Create Mutations

File: app/graphql/mutations/cinema_hall_mutation.rb (hunk 2 above)
Comment (KMosc) [reply app/graphql/mutations/cinema_hall_mutation.rb]: @wmaciejak I'm gonna add https://github.com/testdouble/standard or use rubocop once I'm done with it all. Gonna do it once I finish task from logic side.

File: app/graphql/mutations/ticket_mutation.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+module Mutations
+    class TicketMutation < BaseMutation
+      argument :title, String, required: false
+      argument :paid, Boolean, required: false
+      argument :password, String, required: false
+      argument :seat, String, required: false
+      argument :cinema_hall_id, ID, required: false
+      argument :movie_id, ID, required: false
```
Comment (wmaciejak) [inline app/graphql/mutations/ticket_mutation.rb]: Those two arguments are required probably

File: app/graphql/mutations/movie_mutation.rb (hunk 3 above)
Comment (KMosc) [reply app/graphql/mutations/movie_mutation.rb]: It is. Gonna fix it.

File: app/graphql/mutations/movie_mutation.rb (hunk 4 above)
Comment (KMosc) [reply app/graphql/mutations/movie_mutation.rb]: Ok.

File: app/graphql/mutations/ticket_mutation.rb (hunk 5 above)
Comment (KMosc) [reply app/graphql/mutations/ticket_mutation.rb]: Right.

File: bin/rails [hunk 6]
```diff
@@ -1,4 +1,9 @@
 #!/usr/bin/env ruby
+begin
```
Comment (wmaciejak) [inline bin/rails:2]: Why this code is needed?

**Comment by wmaciejak** [comment]
What kind of problems did you solve here with Spring?

File: app/graphql/mutations/cinema_hall_mutation.rb (hunk 2 above)
Comment (wmaciejak) [reply app/graphql/mutations/cinema_hall_mutation.rb]: Okay 👍 Just remember for the future that it's better to standardize code before you start coding (or at least before code review).

**Comment by KMosc** [comment]
Well there was smth with visual studio autoloading, that's why I was looking for answer to solve it.
That thing wasn't what I wanted, I forgot git restore and pushed, It doesnt matter.

**Comment by KMosc** [comment]
Changelog 10:00am, 9.07.2021
- [x] Change mutations' names.
- [x] Add apollo upload server, Graphql upload works fine.

**Pull request merged**